conan_provider.cmake

```diff
@@ -514,4 +537,8 @@ endmacro()
 
 # Add a deferred call at the end of processing the top-level directory
 # to check if the dependency provider was invoked at all.
-cmake_language(DEFER DIRECTORY "${CMAKE_SOURCE_DIR}" CALL conan_provide_dependency_check)
\ No newline at end of file
+cmake_language(DEFER DIRECTORY "${CMAKE_SOURCE_DIR}" CALL conan_provide_dependency_check)
+
+# Configurable variables for Conan profiles
+set(CONAN_HOST_PROFILE "autodetect" CACHE STRING "Conan host profile")
```

Comment:
The ``autodetect`` name might be a bit confusing for users knowing about ``conan profile detect`` command. Maybe ``cmakedetect``? Don't love it either, ``autodetect`` might be good...

Reply:
maybe just the word `auto`? - so that it's not confused with `conan profile detect`

Reply:
Sounds good, I don't think that many users will have a profile named "auto"?

Reply:
Going for `auto-cmake` which I think is less likely than `auto`!